source/agora/common/Deserializer.d

```diff
@@ -286,7 +288,9 @@ private T deserializeVarInt (T) (scope DeserializeDg dg)
         return cast(T)value;
     }
 
-    if (int_size <= 0xFC)
+    static if (is(Unqual!T == ubyte))
```

Comment:
Did you add this to evade a cast? But wouldn't the compiler already know that a cast from ubyte to ubyte is just a noop? Curious.

Reply:
No, i actually wrote it because `ubyte` are not size-encoded (it doesn't make sense). But to keep the code generic, this function is always called. Come to think of it, modifying the call site might be better...